### PR TITLE
feat: progress bar in affiliates page

### PR DESCRIPTION
**Pull request by tinaszheng**
image
image
image
image

### CODE REVIEW

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  return (
+    <div className={className} tw="flex flex-col gap-1.5 rounded-1 p-1">
+      <div tw="flex items-center justify-between gap-0.5">
+        <div tw="flex flex-col gap-0.375">
+          <div tw="font-semibold">{stringGetter({ key: STRING_KEYS.BECOME_AN_AFFILIATE })}</div>
+          <div tw="text-color-text-0">
+            {stringGetter({ key: STRING_KEYS.TRADE_MORE_VOLUME_REQUIREMENT })}
+          </div>
+        </div>
+        <div tw="text-large font-medium">{progressPercent}%</div>
```
Comment (jaredvu) [inline src/views/Affiliates/cards/AffiliateProgressCard.tsx:28]: `<Output tw="inline" type={OutputType.Percent} value={progressPercent} fractionDigits={0} />` and can remove the `* 100` in `progressPercent`.

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      </div>
+      <div tw="flex flex-col gap-0.75">
+        <div tw="flex justify-between">
+          <div tw="flex items-end">
+            <Output
+              value={volume}
+              type={OutputType.Fiat}
+              slotRight={` ${stringGetter({ key: STRING_KEYS.TRADED })}`}
+            />
```
Comment (jaredvu) [inline src/views/Affiliates/cards/AffiliateProgressCard.tsx:37]: after you get `{VOLUME_USD} traded` string into localization you can
```
{stringGetter({ key: STRING_KEYS.TRADED, params: { VOLUME_USD: <Output ... />}})}
```

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          </div>
+          <div tw="text-color-text-0">
+            <Output
+              value={remaining}
+              type={OutputType.Fiat}
+              slotRight={` ${stringGetter({ key: STRING_KEYS.REMAINING })}`}
+            />
```
Comment (jaredvu) [inline src/views/Affiliates/cards/AffiliateProgressCard.tsx:44]: same

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          </div>
+        </div>
+        <div tw="relative h-2.5 w-full overflow-hidden rounded-0.5 bg-color-layer-7">
+          <div tw="absolute h-full bg-color-accent" style={{ width: `${progressPercent}%` }} />
```
Comment (jaredvu) [inline src/views/Affiliates/cards/AffiliateProgressCard.tsx:48]: ah this is probably where you would have `progressPercent * 100` if you make the above changes

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx (hunk 1 above)
Comment (jaredvu) [reply src/views/Affiliates/cards/AffiliateProgressCard.tsx:28]: positives would be no need to `.toString()` or `.toFixed()`

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+export const AffiliateProgressCard = ({
+  volume = 0,
+  className,
+}: {
+  volume?: number;
+  className?: string;
+}) => {
+  const stringGetter = useStringGetter();
+
+  const progressPercent = ((volume / AFFILIATES_REQUIRED_VOLUME_USD) * 100).toFixed(0).toString();
```
Comment (jaredvu) [inline src/views/Affiliates/cards/AffiliateProgressCard.tsx:17]: isn't `toString()` redundant after `toFixed()`

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx (hunk 1 above)
Comment (tinaszheng) [reply src/views/Affiliates/cards/AffiliateProgressCard.tsx:28]: omg isnt this so much more verbose 💀

Comment (jaredvu) [reply src/views/Affiliates/cards/AffiliateProgressCard.tsx:28]: more verbose but keeps it consistent w/ other number outputs. i.e. mono font-feature-settings, not needing to add `%` suffix, rounding mode, etc.

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx (hunk 5 above)
Comment (tyleroooo) [reply src/views/Affiliates/cards/AffiliateProgressCard.tsx:17]: USE OUTPUT and the associated formatting function

File: src/views/Affiliates/cards/AffiliateProgressCard.tsx (hunk 1 above)
Comment (tinaszheng) [reply src/views/Affiliates/cards/AffiliateProgressCard.tsx:28]: omg im so against this but ill do it

**Review by jaredvu — APPROVED** [review]

**Pull request merged**